frontend/src/pages/modelCatalog/screens/RegisterCatalogModel.tsx

```diff
@@ -192,14 +192,17 @@ const RegisterCatalogModel: React.FC = () => {
 
   return (
     <ApplicationsPage
-      title={`Register ${params.modelName || ''} model`}
+      title={`Register ${decodedParams.modelName || ''} model`}
       description="Create a new model and register the first version of your new model."
       breadcrumb={
         <Breadcrumb>
           <BreadcrumbItem render={() => <Link to="/modelCatalog">Model catalog</Link>} />
           <BreadcrumbItem
+            data-testid="breadcrumb-model-name"
             render={() => (
-              <Link to={getCatalogModelDetailsUrl(params)}>{params.modelName || 'Loading...'}</Link>
+              <Link to={getCatalogModelDetailsUrl(decodedParams)}>
```

Comment:
I wonder if we should bypass the `<Link>` if we are still in a loading state?

`{!params.modelName ? 'Loading...' : <Link... }`

but it wasn't part of your change, just an observation... otherwise lgtm

Reply:
yeah right, we do need to fix this in other places as well. I can address this as part of this issue.